spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -638,6 +650,21 @@ else if (referencePersistentProperty.isCollectionLike()) {
 			}
 	}
 
+	private Object fetchReferenced(DatastorePersistentProperty referencePersistentProperty, ReadContext context, Set<Key> keys) {
```

Comment:
Not too clear to me the difference between `fetchReferenced` and `findReferenced` maybe this requires a comment or renaming.